### ISSUE TITLE
Record context within a field

### ISSUE THREAD

**Issue by naps62** [thoughtbot/administrate#391]
Is it possible to have a custom field whose presentation logic depends on the record that holds it?

I was going through the documentation and the code, but from what I could see, the field has no knowledge of the record it's in

To clarify, I have a model where each record has a different monetary value & currency. I want to do two things:

1. Show the currency symbol alongside the value (this could be done in a separate column, but that's not ideal)
2. Show a different number of decimal places, depending on the currency (so Dollars or Euros would have 2 decimal places, but Bitcoins would have 8)

**Comment by jrguenther** [comment]
Similar to #438

**Comment by andrewhamon** [comment]
Hello @naps62 and @jrguenther!

Check out my response to https://github.com/thoughtbot/administrate/issues/421

tl;dr there is nothing stopping you from defining a method on your model and using it as a "field" in Administrate.

I did [run in to some issues](https://github.com/thoughtbot/administrate/issues/445) though that you should be aware of. My current workaround was to create a [custom field type](http://administrate-prototype.herokuapp.com/adding_custom_field_types) that is identical to `Field::String`, but with with the field defined as not searchable.

**Comment by naps62** [comment]
@andrewhamon thanks for the help. I wasn't thinking about this in the correct way. The `value` field I have on my model is already an instance `Money` (from the [`money`](https://github.com/RubyMoney/money-rails) gem, but I was treating it as a raw number. Knowing it's a Money value, I can just do:

```
class MoneyField < Administrate::Field::Base
  def to_s
    data.format
  end
end
```
